pkg/kubelet/dockershim/helpers_windows.go

```diff
@@ -71,14 +73,35 @@ func (ds *dockerService) determinePodIPBySandboxID(sandboxID string) string {
 		if err != nil {
 			continue
 		}
-		if containerIP := getContainerIP(r); containerIP != "" {
-			return containerIP
+
+		// On Windows, every container that is created in a Sandbox, needs to invoke CNI plugin again for adding the Network,
+		// with the shared container name as NetNS info,
+		// which is passed down the platform to replicate some necessary changes on to the new container
+		//
+		// This place is chosen as a hack for now, until we have a better place to do this,
+		// like immediately after ContainerCreation,
+		// Also this is very windows specific for now
+
+		// Using the CONTAINER_NETWORK flag to determine to choose old impl or current impl.
+		// CONTAINER_NETWORK is only kept for backward compatibility, since old OS version doens;t support
+		// Sandbox concept
+		if networkMode := os.Getenv("CONTAINER_NETWORK"); networkMode == "" {
```

Comment:
weird of using a environment variable to determine kubelet's behavior, which is actually depending on the OS version. Is it possible of getting os version instead?

Reply:
Yes, will fix it separately. In my TODO list.

Reply:
Can you at least document what version relies on this env variable?